docs/developers/contributing/documentation/docs_deployment.md

```diff
@@ -16,36 +16,35 @@ through several CI workflows detailed below.
     - **Workflow file:** [`build_docs.yml`](https://github.com/napari/napari/blob/main/.github/workflows/build_docs.yml)
         - **job:** `build-and-upload`
         - Pulls in sources from `napari/docs` and builds docs locally. Uploads
-          artifacts to this repo (`napari/napari`).
-        - This is triggered on every Pull Request and shows up as a "Build PR
-          Docs" check on the PR.
+          artifacts to this repository (`napari/napari`).
+        - This is triggered on any commit tagged 'v*'.
```

Comment:
So `napari/napari` `build_docs.yml`:

```yml
  push:
    branches:
      - docs
    tags:
      - 'v*'
```

* we don't have a `docs` branch in `napari/napari` (anymore?)?
* we build and upload the artifact, but I can't workout who/when this is used? The `napari/napari` deploy workflow just triggers the `napari/docs` `build_and_deploy` ? I am guessing by the trigger on version tag that maybe this is used for stable version of docs?

(on that note, we state that `napari/docs` `build_and_deploy.yml` always deploys to the `dev/` folder. But how do the 'vxxx' folders get populated??)

This is confusing! :sweat_smile: 

cc @psobolewskiPhD @melissawm

Reply:
Nevermind, I can see how the version docs get uploaded. Then I really don't understand what `napari/napari` `build_docs.yml` workflow is doing, who uses that artifact?

Reply:
napari/napari `build_docs` doesn't run on PRs anymore, we rely on CircleCI. I wasn't sure if that was a conscious effort in the napari/napari CI overhaul, to save runners, or what, but it looks like it was an accident: in the syncing of workflows the branch name was also ported over so it's just never triggered.

I *dont* even think it's used when releasing docs from a branch (like 0.4.19 didn't use the napari/docs repo, but instead cherry picked over to napari/napari), but I'm not sure.
So I think we could delete it, or make it not run on PR so someone could trigger manually if they wanted.

Reply:
I think this is a leftover from before the repo split. Unfortunately we have lived in a split state for a while (some work depends on the monorepo setup, some work depends on the two-repo setup) and so a lot of it is very confusing. I've done my best to try and capture the changes but the repo does change faster than I can keep track of I'm afraid 😢

Reply:
So I took a look at the history and it turns out my first instinct was correct:
the `pull_request` part was removed when napari/napari CI was reworked:
https://github.com/napari/napari/pull/6156/files#diff-4dfe935b4852393622c2b414783d1d11e1a868fd8485709124d90208fa61ec8b

Also, it can be triggered manually, so it's fine to stay I think.

Reply:
And reading more carefully, branch `docs` and tag `v.` would work with releases from branches, like 0.4.19. So we do need to keep it.

Reply:
> And reading more carefully, branch docs and tag v. would work with releases from branches, like 0.4.19. So we do need to keep it.

Interesting. Can you expand? Is a `docs` branch added to `napari/napari` when we release from branches?
And how is the artifact finally used?

Reply:
I'll add this info to the docs :)

Reply:
I didn't phrase that well. That workflow would work in a branch that is before the repo split, so then there is `docs` in napari/napari. See: v0.4.19:
https://github.com/napari/napari/tree/v0.4.19x
So for example this PR:
https://github.com/napari/napari/actions/runs/7703963871

Reply:
Ohhh...I'm not so familiar with releasing from a branch so I haven't followed on some parts.

So when releasing from a branch, you have to make a `vxx` branch AND a `docs_vxx` branch? And the `napari/napari` `build_docs` will run on pushes to `docs*` branches and `v*` tags?

And is the `build_docs` just used to view the docs to make sure they look okay? Are they used for deployment?

Reply:
@psobolewskiPhD I've updated the docs to what I think is happening. Please let me know if something is off, thanks!

Reply:
> So when releasing from a branch, you have to make a vxx branch AND a docs_vxx branch? And the napari/napari build_docs will run on pushes to docs* branches and v* tags?

I'm not sure this is all required, but the write up seems fine now.
Hopefully that workflow never gets used again because we release 0.5.0